src/sinks/elasticsearch/config.rs

```diff
@@ -131,6 +134,12 @@ impl BulkConfig {
     }
 }
 
+#[derive(Deserialize, Serialize, Clone, Debug, Default)]
+pub struct DistributionConfig {
+    pub endpoints: Vec<String>,
```

Comment:
It could, there is already a layer for that in `serde` file `OneOrMany`.

I see this as a UX decision, an intentional paper cut to make users more aware that they are activating distribution, since its activation changes the behavior of this sink in a few ways:
* There is the issue with ARC which is turned on by default.
* On first glance it's unclear how `rate_limit` applies when multiple endpoints are specified so I want to nudge users to reading how distribution works to avoid any "gotcha".

There are also some minor benefits:
* Health check is performed on a single endpoint at the moment so having that endpoint in its own `endpoint` option gives a hint that its special in some way.
* Options regarding distribution are bundled together.

That said, having a single `endpoint` that can accept multiple values and documenting there that if they are specified distribution will be activated, seems fine. We could also rename it to `endpoints` and have `endpoint` as an alias.

Reply:
I think we'd want to go with `endpoint` being a one-or-many, with a documentation upgrade to explain the difference. It's not like people can accidentally pass in multiple hosts to `endpoint` right now, so they would be looking at the documentation to see that this is now possible.

And if they happened to blindly do so, not knowing how the load balancing worked.... dunno, that's not a class of problem we can solve with the naming of configuration values. 🤷🏻

Reply:
True